operations/host_provision.go

```diff
@@ -0,0 +1,127 @@
+package operations
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/evergreen-ci/evergreen/cloud/userdata"
+	"github.com/evergreen-ci/evergreen/rest/client"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/jasper"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+func hostProvision() cli.Command {
+	const (
+		hostIDFlagName       = "host_id"
+		hostSecretFlagName   = "host_secret"
+		workingDirFlagName   = "working_dir"
+		apiServerURLFlagName = "api_server"
+	)
+	return cli.Command{
+		Name:  "provision",
+		Usage: "fetch and run the host provisioning script",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  hostIDFlagName,
+				Usage: "the host ID",
+			},
+			cli.StringFlag{
+				Name:  hostSecretFlagName,
+				Usage: "the host secret",
+			},
+			cli.StringFlag{
+				Name:  workingDirFlagName,
+				Usage: "the working directory for the script",
+			},
+			cli.StringFlag{
+				Name:  apiServerURLFlagName,
+				Usage: "the base URL for the API server",
+			},
+		},
+		Before: mergeBeforeFuncs(
+			requireStringFlag(hostIDFlagName),
```

Comment:
Can't you use the required field in the flag?

Reply:
We can't because it would require an upgrade to urfave/cli, which requires go1.10+ after the required flag was introduced. Most of the require funcs in the CLI package could go away if we upgraded our go version.

Reply:
wonderful lol